### ISSUE TITLE
Switch to using sequence-based checksums for sequence identifiers so they will be comparable across runs

### ISSUE THREAD

**Issue by stenglein-lab** [stenglein-lab/tick_surveillance#32]
Although in theory the sequences themselves allow comparison across runs this will create sequence identifiers (names) that are (1) shorter than the actual sequences and (2) facilitate sequence comparison across runs.  

This will require that we confirm that the very low probability event of a collision where two non-identical sequences produce the same hash

**Comment by stenglein-lab** [comment]
This change was introduced in 3f79f91.   Once the dev branch is merged into main this can be closed.

### A note about hash collisions

There is a *very* low probability that two different sequences could produce the same SHA1 checksum.  This is called a "hash collision" and these are described on this page:

https://en.wikipedia.org/wiki/Hash_collision

This says: "For a SHA-1 function to have a 50% chance of a hash collision occurring, there would have to be 1.42 × 10^24 records..."

10^24 is a huge number, so this is very very unlikely to ever occur with the small # of sequences that will show up in MPAS.  But downstream analysis tools (cross-run comparisons, etc.) should check for the possibility of a collision, which would manifest as two sequences with the same SHA1 checksum, but different actual sequences.
